geode-core/src/main/java/org/apache/geode/cache/query/internal/AttributeDescriptor.java

```diff
@@ -142,32 +146,44 @@ Member getReadMember(ObjectType targetType) throws NameNotFoundException {
   }
 
   Member getReadMember(Class targetClass) throws NameNotFoundException {
+
     // mapping: public field (same name), method (getAttribute()),
     // method (attribute())
     List key = new ArrayList();
     key.add(targetClass);
     key.add(_name);
 
-    Member m = (Member) _cache.get(key);
-    if (m != null)
+    Member m = (Member) _local_field_cache.get(key);
```

Comment:
This code looks good - but all of this probably could be written more cleanly with using one cache and a compute method. Eg:

Member m = localCache.computeIfAbsent(key, this::getReadMember)

getReadMember(Class target) {
   Member m = getReadField(target)
   return m == null ? getReadMethod(target) : m;
}